apiserver/facades/client/application/application_test.go

```diff
@@ -1953,9 +1984,9 @@ func (s *applicationSuite) TestApplicationUpdateAllParams(c *gc.C) {
 	// Check the minimum number of units.
 	c.Assert(app.MinUnits(), gc.Equals, minUnits)
 
-	// Check the settings: also ensure the YAML settings take precedence
-	// over strings ones.
-	expectedSettings := charm.Settings{"blog-title": "yaml-title"}
+	// Check the settings: also ensure the string settings take precedence
```

Comment:
This contradicts what we see in the deploy code [here](https://github.com/juju/juju/blob/2.8/apiserver/facades/client/application/application.go#L681-L713). AFAICT, the string settings override the YAML ones. 

I have changed the test to match what we actually see in the code.

@wallyworld Can you clarify whether this is the expected behavior?

Reply:
I think the code comments are wrong.

```
type ApplicationUpdate struct {
	SettingsStrings map[string]string  `json:"settings,omitempty"`
	SettingsYAML    string             `json:"settings-yaml"` // Takes precedence over SettingsStrings if both are present.
```

```
type ApplicationDeployV6 struct {
	Config           map[string]string              `json:"config,omitempty"`
	ConfigYAML       string                         `json:"config-yaml"` // Takes precedence over config if both are present.
```

In both cases, the YAML is processed first and the config key values then takes precedence.